evm/exceptions.py

```diff
@@ -141,3 +149,25 @@ class OutOfBoundsRead(VMError):
     boundaries of the buffer (such as with RETURNDATACOPY)
     """
     pass
+
+
+class UnannouncedStateAccess(VMError):
+    """
+    Error raised if a storage slot or account field is accessed that has not
+    been specified in the transaction's read or write list, respectively.
+    """
+    pass
+
+
+class GasPriceAlreadySet(Exception):
```

Comment:
@NIC619  Should it be a `VMError`?
```python
class GasPriceAlreadySet(Exception):
```

Reply:
I think it's intentionally not a VM error to ensure that mis-use of this API results in a hard failure, requiring EVM code that uses this API to correctly handle all of the cases (such as the price already being set).